unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ChannelLinkDetector.cs

```diff
@@ -0,0 +1,98 @@
+using DCL.Chat.Channels;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace DCL.Chat.HUD
+{
+    public class ChannelLinkDetector : MonoBehaviour, IPointerClickHandler
```

Comment:
Is this required to be a Monobehaviour?

Reply:
It is required because its set in the game objects that contains a TextMeshPro component. It also contains OnPointerClickHandler behaviour